control_v2.proto

```diff
@@ -259,4 +271,12 @@ service ElasticAgentControl {
 
   // Gather diagnostic information for the running units.
   rpc DiagnosticUnits(DiagnosticUnitsRequest) returns (stream DiagnosticUnitResponse);
+
+  // Configure adjusts the running Elastic Agent configuration with the configuration
+  // provided over the RPC.
+  //
+  // This is only allowed if the Elastic Agent is spawned in TESTING_MODE. Calling this
+  // on any Elastic Agent that is not in TESTING_MODE will result in an error being
+  // returned and nothing occurring.
+  rpc Configure(ConfigureRequest) returns (Empty);
```

Comment:
I'm a bit late to the party and probably miss some context but what's the difference between providing the configuration to the agent via GRPC and the normal Fleet or standalone (reloadable) config the agent normally use ?

If this is preparation for e2e or integration tests shouldn't we test the agent operating like it would in production?

If we wanna trigger config reload from files or restarts on demand, wouldn't a signal handler be a simpler solution ?

Reply:
The integration testing framework is for testing the Elastic Agent inside of its own repository as much as it is testing components from other repositories. This `Configure` rpc does not have the requirement  that it always be used for the integration testing. We can test reading from configuration files as well, but the `Configure` makes it much easier to get status back from the Elastic Agent when a configuration is bad and report it directly to a test.

Also testing flows from Fleet require a full elasticsearch, Kibana, and Fleet Server, that is not really an integration test and more of an end-to-end test.

Reply:
If I understand correctly, what we want to have as a setup is an Agent running and some fake binaries interacting with the Agent inputs and watching the Agent status, correct?

I do not mind a `StateWatch` at all, I was just concerned that we were cutting off a part of the agent functionality by only using something like the `Configure` endpoint